src/gsy_e/models/strategy/heat_pump.py

```diff
@@ -0,0 +1,171 @@
+from typing import Dict, TYPE_CHECKING
+
+from gsy_framework.constants_limits import ConstSettings
+from gsy_framework.data_classes import Trade
+from gsy_framework.enums import AvailableMarketTypes
+from pendulum import DateTime, duration
+
+from gsy_e.constants import FLOATING_POINT_TOLERANCE
+from gsy_e.constants import FutureTemplateStrategiesConstants
+from gsy_e.models.state import HeatPumpState
+from gsy_e.models.strategy.energy_parameters.heat_pump import HeatPumpEnergyParameters
+from gsy_e.models.strategy.new_base_strategy import NewStrategyBase
+from gsy_e.models.strategy.order_updater import OrderUpdaterParameters, OrderUpdater
+
+DEFAULT_HEAT_PUMP_ORDER_UPDATE_PARAMS = {
+    AvailableMarketTypes.SPOT: OrderUpdaterParameters(
+        initial_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.initial,
+        final_rate=ConstSettings.HeatPumpSettings.BUYING_RATE_RANGE.final,
+        update_interval=duration(
+                minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)),
+    AvailableMarketTypes.FUTURE: OrderUpdaterParameters(
+        initial_rate=FutureTemplateStrategiesConstants.INITIAL_BUYING_RATE,
+        final_rate=FutureTemplateStrategiesConstants.FINAL_BUYING_RATE,
+        update_interval=duration(
+            minutes=FutureTemplateStrategiesConstants.UPDATE_INTERVAL_MIN)),
+}
+
+if TYPE_CHECKING:
+    from gsy_e.models.market import MarketBase
+
+
+class HeatPumpStrategy(NewStrategyBase):
+    """Strategy for heat pumps with storages."""
+    # pylint: disable=too-many-arguments)
+    def __init__(self,
+                 maximum_power_rating: float = ConstSettings.HeatPumpSettings.MAX_POWER_RATING_KW,
+                 min_temperatur_C: float = ConstSettings.HeatPumpSettings.MIN_TEMP_C,
+                 max_temperatur_C: float = ConstSettings.HeatPumpSettings.MAX_TEMP_C,
+                 external_temperatur_C: float = ConstSettings.HeatPumpSettings.EXT_TEMP_C,
+                 tank_volume_l: float = ConstSettings.HeatPumpSettings.TANK_VOL_L,
+                 average_consumption: float =
+                 ConstSettings.HeatPumpSettings.AVERAGE_CONSUMPTION_KW,
+                 source_type: int = ConstSettings.HeatPumpSettings.TYPE,
+                 order_updater_parameters: Dict[
+                     AvailableMarketTypes, OrderUpdaterParameters] = None,
+                 preferred_buying_rate: float =
+                 ConstSettings.HeatPumpSettings.PREFERRED_BUYING_RATE
+                 ):
```

Comment:
Please ignore the interface for now. It will slightly change with GSYE-426. Thanks!